### DISCUSSION TITLE
aiomysql

### DISCUSSION THREAD

**Discussion by pous-sin** [flet-dev/flet#3280] [Packages] [answered]
.

**Comment by ndonkoHenri** [comment, marked as answer]
**aiomysql works on Flet mobile — no special build or recipe needed.** It's pure-Python (built on PyMySQL), so `flet build` bundles it straight from PyPI.

I verified it on-device on both an **Android arm64 emulator** and an **iOS simulator**: `import` works and a full `CREATE` / `INSERT` / `SELECT` round-trip against a MariaDB 11.8 server succeeds on both. 

Two things worth knowing:

**1. It's a network client.** It connects to a *remote* MySQL/MariaDB server — there's no embedded database on the phone. With `mysql_native_password` / MariaDB, no extra dependency is needed. For MySQL 8's default `caching_sha2_password` over a non-TLS connection you'll also need `cryptography` (which Flet already [supports](https://pypi.flet.dev/cryptography)) or just connect over TLS.

**2. On iOS, add a one-line shim.** Flet's iOS Python runtime doesn't ship the Unix-only `pwd` stdlib module, and aiomysql calls `getpass.getuser()` at import time guarded only by `except KeyError`. So on iOS `import aiomysql` crashes with `ModuleNotFoundError: No module named 'pwd'` (Android is unaffected — it ships `pwd`). Set a username env var *before* importing and it imports cleanly:

```py
import os
os.environ.setdefault("LOGNAME", os.environ.get("USER") or "fletuser")

import aiomysql  # imports fine on iOS now
```

This is an upstream aiomysql quirk (tracked in aio-libs/aiomysql#1053). Note the in-flight [fix](https://github.com/aio-libs/aiomysql/pull/1073) there only broadens the guard to `except (KeyError, OSError)`, which still won't cover Flet's case — flet currently bundles Python 3.12, where the missing-`pwd` error is a raw `ImportError`, not `OSError`. The complete fix would match PyMySQL's `except (ImportError, KeyError, OSError)`. Until that lands, the env-var shim above is all you need.
